carmm/analyse/meshgrid/meshgrid_functions.py

```diff
@@ -0,0 +1,153 @@
+        y_vec_dist = np.where(y_vec_dist > 0.5, y_vec_dist - 1., y_vec_dist)
+    if MeshObject.pbc[2]==1:
+        z_vec_dist = np.where(z_vec_dist > 0.5, z_vec_dist - 1., z_vec_dist)
+
+    # Convert mesh distances back to cartesian coordinates.
+    stack_mesh = np.stack((x_vec_dist, y_vec_dist, z_vec_dist), axis=-1)
+    stack_mesh = np.einsum('ji,abcj->jabc', MeshObject.Cell.array, stack_mesh)
+
+    mesh_distances = np.linalg.norm(stack_mesh, axis=0)
+
+    return mesh_distances
+
+def distance_point2point(x_1, y_1, z_1, x_2, y_2, z_2, MeshObject):
+    """
+    Function finds the distance between two points (defined in cartesian co-ordinates).
+    Args:
+        x_1: float
+            x coordinate of point 1.
+        y_1: float
+            y coordinate of point 1.
+        z_1: float
+            z coordinate of point 1.
+        x_2: float
+            x coordinate of point 2.
+        y_2: float
+            y coordinate of point 2.
+        z_2: float2
+            z coordinate of point 2.
+        MeshObject: MeshgridObject
+            Object storing meshgrid and PBC conditions
+    Returns:
+        o_distance: Distance between points 1 and 2.
+    """
+
+    from ase.geometry import get_distances
+
+    o_distance = get_distances([x_1, y_1, z_1], p2=[x_2, y_2, z_2],
+                                cell=MeshObject.Cell, pbc=MeshObject.pbc)
+
+    return o_distance
+
+def midpoint_points(x_1, y_1, z_1, x_2, y_2, z_2, MeshObject):
+    """
+    Function finds the distance between two points (defined in cartesian co-ordinates).
+    Args:
+        x_1: float
+            x coordinate of point 1.
+        y_1: float
+            y coordinate of point 1.
+        z_1: float
+            z coordinate of point 1.
+        x_2: float
+            x coordinate of point 2.
+        y_2: float
+            y coordinate of point 2.
+        z_2: float2
+            z coordinate of point 2.
+        MeshObject: MeshgridObject
+            Object storing meshgrid and PBC conditions
+    Returns:
+        x_mid: float
+            x coordinate of midpoint between points 1 and 2.
+        y_mid: float
+            y coordinate of midpoint between points 1 and 2.
+        z_mid: float
+            z coordinate of midpoint between points 1 and 2.
+    """
+
+    import numpy as np
+    from ase.geometry import find_mic
+
+    vec1 = np.array([x_1, y_1, z_1])
+    vec2 = np.array([x_2, y_2, z_2])
+
+    mic_shift = find_mic((vec2 - vec1), cell=MeshObject.Cell.array)
+
+    midpoint = np.linalg.norm(vec1+(mic_shift/2))
+
+    return midpoint
+
+def atom_mesh_build_mask(MeshObject, Atom):
+    """
+    Defines meshgrid of based on a radius set by atomic_radius. Uses 99.99 as a junk value.
+    FUTURE DEVELOPMENT:
+    - Replace junk value with None??
+    Args:
+        MeshObject: MeshgridObject
+            Object storing meshgrid and PBC conditions
+        Atom: Atoms object
+            Contains information (atomic symbols and positions)
```

Comment:
There is an Atom class in ASE, the name can be confusing, change to atoms or Atoms

Reply:
Quite right. Resolved.